### PR TITLE
Add new login page design

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr implements the new login page. Huge thanks to @darmiel for the amazing design <3

### CODE REVIEW

**Review by memeToasty — APPROVED** [review]

**Review by TomRomeo — APPROVED** [review]
🫶🏻

File: frontend/pages/login.tsx [hunk 1]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+								<span className="whitespace-nowrap">
+									{login ? (
+										<Trans>
+											Don&apos;t have an account?
+										</Trans>
+									) : (
+										<Trans>Already have an account?</Trans>
+									)}
+								</span>
+								<a
```
Comment (darmiel) [inline frontend/pages/login.tsx:244]:
```suggestion
								<button
```

File: frontend/pages/login.tsx [hunk 2]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+										setLogin((prev) => !prev);
+									}}
+									onKeyUp={(event) => {
+										if (event.key === "Enter") {
+											event.target.dispatchEvent(
+												new Event("click", {
+													bubbles: true,
+												})
+											);
+										}
```
Comment (darmiel) [inline frontend/pages/login.tsx:257]: I think you introduced a function for this a few PRs ago

File: frontend/pages/login.tsx [hunk 3]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+									}}
+									tabIndex={0}
+								>
+									{login ? (
+										<Trans>Sign up now!</Trans>
```
Comment (darmiel) [inline frontend/pages/login.tsx:262]:
```suggestion
										<Trans>Sign Up now!</Trans>
```

File: frontend/pages/login.tsx [hunk 4]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+									) : (
+										<Trans>Sign in now!</Trans>
```
Comment (darmiel) [inline frontend/pages/login.tsx:264]:
```suggestion
										<Trans>Sign In now!</Trans>
```

File: frontend/pages/login.tsx [hunk 5]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+												<Trans>
+													Passwords have to match
+												</Trans>
+											</Text>
+										</div>
+									</div>
+								)}
+								<Button
+									id="loginSubmitButton"
+									buttonIcon="ArrowRight"
```
Comment (darmiel) [inline frontend/pages/login.tsx:209]: why pass the icon as a string?

File: frontend/pages/login.tsx [hunk 6]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+											<Text
+												textSize="5xs"
+												className="font-light text-[#676767]"
+											>
+												<Trans>
+													Minimum {passwordMinLength}{" "}
+													characters
+												</Trans>
+											</Text>
+										</div>
```
Comment (darmiel) [inline frontend/pages/login.tsx:186]: extract into local component

File: frontend/pages/login.tsx [hunk 7]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+												);
+											}
+											setPasswordsMatching(
+												passwordInput.current?.value ===
+													event.target.value
+											);
+										}}
+									/>
+								)}
+								{!login && (
```
Comment (darmiel) [inline frontend/pages/login.tsx:163]: these can be me<>rged</>

File: frontend/pages/login.tsx [hunk 8]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+					<div className="flex h-full w-full items-center justify-center bg-gradient-to-tr from-[#83DAFF]/20 p-3 sm:p-5">
+						<div className="flex w-80 flex-col items-center space-y-3 rounded-md bg-white p-5 shadow-[0_35px_60px_-15px_rgba(0,0,0,0.2)] md:px-7">
+							<Logo
+								text={false}
+								logoSize="sm"
+								className="p-3"
 								onClick={() => {
-									emailInput.current?.focus();
-									setLogin((prev) => !prev);
+									router.push("/home");
```
Comment (darmiel) [inline frontend/pages/login.tsx:80]: use `next/link`

File: frontend/pages/login.tsx (hunk 2 above)
Comment (Julian702) [reply frontend/pages/login.tsx:257]: No, it's still in our backlog and will be implemented in a future pr.

File: frontend/pages/login.tsx (hunk 8 above)
Comment (Julian702) [reply frontend/pages/login.tsx:80]: It's already part of the backlog (#197)

File: frontend/pages/login.tsx [hunk 9]
```diff
@@ -65,157 +68,210 @@ export default function Page() {
...
+									onClick={() => {
+										if (login) {
+											loginLogic()
+												.then((result) => {})
+												.catch((error) => {});
+										} else {
+											registerLogic()
+												.then((result) => {})
+												.catch((error) => {});
+										}
```
Comment (4KevR) [inline frontend/pages/login.tsx:221]: Maybe refactor to either delete the promise resolution (I know Sonarcloud has a problem with this) or find another way to implement the logic separation.

File: frontend/pages/login.tsx (hunk 3 above)
Comment (Julian702) [reply frontend/pages/login.tsx:262]: @4KevR and I would prefer the lowercase variant.

**Review by 4KevR — CHANGES_REQUESTED** [review]
Great work!

File: frontend/pages/login.tsx (hunk 4 above)
Comment (Julian702) [reply frontend/pages/login.tsx:264]: s.a.

File: frontend/pages/login.tsx (hunk 1 above)
Comment (Julian702) [reply frontend/pages/login.tsx]: fixed in d6f6fb8dd2f10f19a56fb12858f2a7ffb318ef84

File: frontend/pages/login.tsx (hunk 5 above)
Comment (Julian702) [reply frontend/pages/login.tsx]: fixed in d6f6fb8dd2f10f19a56fb12858f2a7ffb318ef84

File: frontend/pages/login.tsx (hunk 7 above)
Comment (Julian702) [reply frontend/pages/login.tsx]: fixed in d6f6fb8dd2f10f19a56fb12858f2a7ffb318ef84

File: frontend/pages/login.tsx (hunk 9 above)
Comment (Julian702) [reply frontend/pages/login.tsx]: fixed in d6f6fb8dd2f10f19a56fb12858f2a7ffb318ef84

**Review by 4KevR — APPROVED** [review]

File: frontend/components/input/Button.tsx [hunk 10]
```diff
@@ -5,6 +5,10 @@ import { Text } from "../Text";
 import { Icon, IconName } from "../graphics/Icon";
 
 export interface ButtonProps {
+	/**
+	 * Text
+	 */
+	buttonText?: string;
```
Comment (darmiel) [inline frontend/components/input/Button.tsx:11]: this is horrible

File: frontend/pages/login.tsx [hunk 11]
```diff
@@ -65,157 +62,193 @@ export default function Page() {
...
+											></PasswordCheck>
+										</div>
+									</>
+								)}
+								<Button
+									id="loginSubmitButton"
+									buttonText={
+										login
+											? _(msg`Sign in`)
+											: _(msg`Register`)
```
Comment (darmiel) [inline frontend/pages/login.tsx:190]:
```suggestion
											: _(msg`Sign up`)
```

File: frontend/pages/login.tsx [hunk 12]
```diff
@@ -65,157 +62,193 @@ export default function Page() {
...
+											loginLogic();
+										} else {
+											registerLogic();
+										}
+									}}
+								>
+									<ArrowRight
+										size={16}
+										className="flex-none"
+									></ArrowRight>
```
Comment (darmiel) [inline frontend/pages/login.tsx:205]:
```suggestion
									/>
```

**Pull request merged**